presto-main/src/test/java/com/facebook/presto/connector/MockConnectorFactory.java

```diff
@@ -207,4 +207,41 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
             }
         }
     }
+
+    public static final class Builder
+    {
+        private Function<ConnectorSession, List<String>> listSchemaNames;
+        private BiFunction<ConnectorSession, String, List<SchemaTableName>> listTables;
+        private BiFunction<ConnectorSession, SchemaTablePrefix, Map<SchemaTableName, ConnectorViewDefinition>> getViews;
+        private BiFunction<ConnectorSession, ConnectorTableHandle, Map<String, TpchColumnHandle>> getColumnHandles;
```

Comment:
You could put some defaults here. That way user could call only methods that relevant to they use case.